reactioncommerce/reaction-component-library
# #98: ProgressiveImage component ported from Storefront

Resolves #98 
Impact: **minor**  
Type: **feature**

## Component
This PR ports over Storefront's Medium/Instagram-like progressive image component.

## Screenshots

##### Responsive image at smaller size
![responsive-small](https://user-images.githubusercontent.com/849570/44111978-da607da0-9fd1-11e8-88c2-394784f610d3.jpg)

##### Responsive image at larger size
<img width="551" alt="responsive-large" src="https://user-images.githubusercontent.com/849570/44111988-e1d0a600-9fd1-11e8-912a-bddea336f065.png">


## Breaking changes
None

## Testing
1. Confirm Image is listed under Storefront Components -> General -> Image
2. Confirm component works with single image (`src`), progressive loading (`presrc`), and responsive (`srcs`)
